updater/spec/support/dummy_pkg_helpers.rb

```diff
@@ -61,4 +62,38 @@ def create_temporary_content_directory(fixture:, directory: "/", state: "origina
   def updated_bundler_files_hash(fixture: "bundler")
     updated_bundler_files(fixture: fixture).map(&:to_h)
   end
+
+  # Stub PackageManagerBase
+  class StubPackageManager < Dependabot::PackageManagerBase
+    def initialize(name:, version:, deprecated_versions: [], unsupported_versions: [], supported_versions: [])
+      @name = name
+      @version = version
+      @deprecated_versions = deprecated_versions
+      @unsupported_versions = unsupported_versions
+      @supported_versions = supported_versions
+    end
+
+    attr_reader :name
+    attr_reader :version
+    attr_reader :deprecated_versions
+    attr_reader :unsupported_versions
+    attr_reader :supported_versions
+
+    sig { override.returns(T::Boolean) }
+    def deprecated?
+      # If the version is unsupported, the unsupported error is getting raised separately.
+      return false if unsupported?
+
+      deprecated_versions.include?(version)
+    end
+
+    sig { override.returns(T::Boolean) }
+    def unsupported?
+      # Check if the feature flag for Bundler v1 unsupported error is enabled.
+      return false unless Dependabot::Experiments.enabled?(:bundler_v1_unsupported_error)
+
+      # Determine if the Bundler version is unsupported.
+      version < supported_versions.first
```

Comment:
🔎 I'm okay with this landing _now_, but I think it's preferable to exclude Bundler-specific concerns in this generic stub.

```suggestion
      # Check if the feature flag for Bundler v1 unsupported error is enabled.
      return false unless name == "bundler" && Dependabot::Experiments.enabled?(:bundler_v1_unsupported_error)

      version < supported_versions.first
```